org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/TypePattern.java

```diff
@@ -276,7 +277,11 @@ boolean subtreeMatch0(ASTMatcher matcher, Object other) {
 	ASTNode clone0(AST target) {
 		TypePattern result = new TypePattern(target);
 		result.setSourceRange(getStartPosition(), getLength());
-		result.setPatternVariable((VariableDeclaration) getPatternVariable().clone(target));
+		if (this.ast.apiLevel < AST.JLS22_INTERNAL) {
+			result.setPatternVariable((VariableDeclaration) getPatternVariable().clone(target));
+		} else {
+			result.setPatternVariable((VariableDeclaration) getPatternVariable2().clone(target));
```

Comment:
cleanup: refer to MethodDeclaration.modifiers - see how it got changed from int to List. Can we think of using a similar approach here

Reply:
Done

Reply:
thanks